stringoruri.go

```diff
@@ -144,10 +145,30 @@ func (s *StringOrURI) UnmarshalCBOR(data []byte) error {
 	return nil
 }
 
-func isCBORTag(data []byte) bool {
-	return (data[0] & 0xe0) == 0xc0
+// MarshalJSON encodes the receiver StringOrURI into a JSON string
+func (s StringOrURI) MarshalJSON() ([]byte, error) {
+	return json.Marshal(s.String())
 }
 
-func isCBORTextString(data []byte) bool {
-	return (data[0] & 0xe0) == 0x60
+// UnmarshalJSON attempts at decoding the supplied JSON data into the receiver
+// StringOrURI
+func (s *StringOrURI) UnmarshalJSON(data []byte) error {
+	var v string
+
+	if err := json.Unmarshal(data, &v); err != nil {
+		return err
+	}
+
+	// RFC 7519, Section 2: any value containing a ":" character MUST be a URI
+	if strings.Contains(v, ":") {
```

Comment:
This check is now done as part of `FromString` and does not need to be explicitly checked here.

Reply:
right-o!